### ISSUE TITLE
Kafka Gateway is not working

### ISSUE THREAD

**Issue by apuigsech** [argoproj/argo-events#485]
**Describe the bug**
A simple Kafka gateway I setup using this definition is not working at all;

```
apiVersion: argoproj.io/v1alpha1
kind: Gateway
metadata:
  name: kafka-gateway
  labels:
    gateways.argoproj.io/gateway-controller-instanceid: argo-events
spec:
  replica: 1
  type: kafka
  eventSourceRef:
    name: kafka-event-source
  template:
    metadata:
      name: kafka-gateway
      labels:
        gateway-name: kafka-gateway
    spec:
      containers:
        - name: gateway-client
          image: argoproj/gateway-client:v0.12
          imagePullPolicy: Always
          command: ["/bin/gateway-client"]
        - name: kafka-events
          image: argoproj/kafka-gateway:v0.12
          imagePullPolicy: Never
          command: ["/bin/kafka-gateway"]
      serviceAccountName: argo-events-sa
  subscribers:
    http:
      - "http://kafka-sensor.argo-events.svc:9300/"
```

The kafka-gateway logs is showing the following;

```
INFO[2020-02-08 00:01:15] started processing the event source...        event-source=kafka-event-source
INFO[2020-02-08 00:01:15] parsing the event source...                   event-source=kafka-event-source
INFO[2020-02-08 00:01:15] connecting to Kafka cluster...                event-source=kafka-event-source
```

That means that the process is getting blocked somewhere when trying to allocate the sarama consumer, without showing any error.

**Additional context**
The issue happens because kafkaEventSource is not properly allocated. Is created as a (nil) pointer, and yaml.Unmarshal cannot allocate it either. This can be fixed doing the following change on gateways/server/kafka/start.go;

From;

```
	var kafkaEventSource *v1alpha1.KafkaEventSource
	if err := yaml.Unmarshal(eventSource.Value, kafkaEventSource); err != nil {
		errorCh <- err
		return
	}
```

To;

```
	var kafkaEventSource v1alpha1.KafkaEventSource
	if err := yaml.Unmarshal(eventSource.Value, &kafkaEventSource); err != nil {
		errorCh <- err
		return
	}
```

If I have some time I will just work on the PR, otherwise you have already the solution here.

**Comment by VaibhavPage** [comment]
I'll put the fix in #486

**Comment by VaibhavPage** [comment]
The fix is out in v0.12.1. Closing the issue for now.
